lms/djangoapps/courseware/views.py

```diff
@@ -1050,6 +1055,21 @@ def _progress(request, course_key, student_id):
     # checking certificate generation configuration
     show_generate_cert_btn = certs_api.cert_generation_enabled(course_key)
 
+    if is_credit_course(course_key):
+        requirement_statuses = get_credit_requirement_status(course_key, student.username)
+        if any(requirement['status'] == 'failed' for requirement in requirement_statuses):
+            eligibility_status = "Not Eligible"
```

Comment:
Correction. These aren't actually printed, so they should be slugs (e.g. eligible, not_eligible, partial_eligibility) to avoid confusion in the future.